modules/@apostrophecms/schema/ui/apos/logic/AposSchema.js

```diff
@@ -269,53 +273,18 @@ export default {
       if (!this.schemaReady) {
         return;
       }
-      const oldHasErrors = this.next.hasErrors;
-      // destructure these for non-linked comparison
-      const oldFieldState = { ...this.next.fieldState };
-      const newFieldState = { ...this.fieldState };
-
-      let changeFound = false;
-
       this.next.hasErrors = false;
-      this.next.fieldState = { ...this.fieldState };
-
       this.schema
         .filter(field => this.displayComponent(field))
         .forEach(field => {
           if (this.fieldState[field.name].error) {
             this.next.hasErrors = true;
           }
-          // This simply check if a field has changed since it has been
-          // instantiated
-          if (
-            this.fieldState[field.name].data !== undefined &&
-            detectFieldChange(
-              field,
-              this.next.data[field.name],
-              this.fieldState[field.name].data
-            )
-          ) {
-            changeFound = true;
-
-            // fieldState never gets the relationships postprocessed data
-            // that's why it gets seen as different than next all the time
-            this.next.data[field.name] = this.fieldState[field.name].data;
-          } else {
-            this.next.data[field.name] = this.modelValue.data[field.name];
-          }
+          this.next.data[field.name] = this.fieldState[field.name].data;
         });
-      if (
-        oldHasErrors !== this.next.hasErrors ||
-        oldFieldState !== newFieldState
-      ) {
-        // Otherwise the save button may never unlock
-        changeFound = true;
-      }
 
-      if (changeFound) {
-        // ... removes need for deep watch at parent level
-        this.$emit('update:model-value', { ...this.next });
-      }
+      this.next.fieldState = { ...this.fieldState };
+      this.$emit('update:model-value', this.next);
```

Comment:
Are those 2 lines equivalent to what we had before with `...this.next`?

Reply:
This line was already there:
```
      this.next.fieldState = { ...this.fieldState };
```

For the emit I removed the spread because it's already done in the parent that spreads `value.data` and is useless I think. (we can keep it if you have a doubt).
```
    updateDocFields(value) {
      this.updateFieldErrors(value.fieldState);
      this.docFields.data = {
        ...this.docFields.data,
        ...value.data
      };
}
```

Reply:
kept the spread